Make the chat looks like actually chat looks like

blukai opened the pull request:
Mobile friendly ✔️, with spam detection & filters

![image](https://user-images.githubusercontent.com/20866892/28795555-fc6b0be0-7642-11e7-9c3d-5f44cf638d01.png)

dartharaknus commented:
Looks great. But I think the filters need to be above the table. Side doesn't work as good.

howardchung commented on src/lang/en-US.json:
```diff
@@ -919,6 +919,14 @@
   "vision_placed_sentry": "placed Sentry at",
   "vision_ward_log": "Ward Log",
 
+  "chat_radiant": "Radiant",
```
I think we can reuse general_dire and general_radiant

howardchung commented on src/components/Match/Chat/Chat.jsx:
```diff
@@ -0,0 +1,225 @@
+    );
+
+    const Filters = () => (
+      <ul className={styles.Filters}>
```
I think these should be labeled as filters

blukai commented:
@howardchung I deleted duplicated by me strings from lang and added heading for filters.
@dartharaknus I'll move the filters above the messages, and then we'll decide how best.

blukai commented:
I replaced switchers by checks since switchers are awful in horizontal placement. Also I was thinking to categorize it, like 'side', 'type', 'other' and replacing `chatwheel` filter by 2 filters: phrases and audio. Btw I like more how it looks on top instead of right. Let me know what u think.

![image](https://user-images.githubusercontent.com/20866892/28819838-1e3b0f98-76b8-11e7-88f1-b09623234e42.png)

![image](https://user-images.githubusercontent.com/20866892/28819898-4e17dd0e-76b8-11e7-8d8a-2a2e3106c6a4.png)

I didn't push commit yet, will wait for reply and then act 🦄

howardchung commented:
It's not immediately clear that the number is a count of that type of message.

blukai commented:
![image](https://user-images.githubusercontent.com/20866892/28978806-256cec06-7950-11e7-9baa-0b7214b37628.png)

howardchung commented on src/components/Match/Chat/Chat.jsx:
```diff
@@ -0,0 +1,294 @@
+      audio: true,
+      all: true,
+      allies: true,
+      spam: false,
```
I think default behavior should be unfiltered.  We just want to show the chat as it was, and allow the user the option to filter if they want?

howardchung commented on src/components/Match/Chat/Chat.jsx:
```diff
@@ -0,0 +1,294 @@
+        type: 'type',
+        disabled: () => this.state.phrases === false && this.state.audio === false,
+      },
+      phrases: {
```
I've actually recently removed phrases from the data (leaving only audio chatwheel messages)

blukai replied:
Hm, anyway I think better to keep this for matches that got this data

howardchung commented on src/lang/en-US.json:
```diff
@@ -296,7 +296,7 @@
   "heading_cosmetics": "Cosmetics",
   "heading_log": "Log",
-  "heading_chat": "Chat",
+  "heading_chat": "Chat messages",
```
I think we should just leave this as "Chat"

howardchung commented on src/lang/en-US.json:
```diff
@@ -341,11 +341,12 @@
-  
+
   "subheading_avg_and_max": "in last %d matches",
+  "subheading_chat": "and categorized filters with counts beside",
```
I don't think this subheading is needed.
